controllers/kbsconfig_controller.go

```diff
@@ -318,91 +371,280 @@ func (r *KbsConfigReconciler) newKbsDeployment(ctx context.Context) *appsv1.Depl
 		} else if err != nil {
-			// Unknown error
-			r.log.Error(err, "Failed to get KBS ConfigMap")
-			return nil
+			r.log.Error(err, "Failed to get KBS Auth Secret")
+			return nil, err
 		}
-		// ConfigMap exists
-		// Create the volume
+
 		volumes = append(volumes, corev1.Volume{
-			Name: "kbs-config",
+			Name: "auth-secret",
 			VolumeSource: corev1.VolumeSource{
-				ConfigMap: &corev1.ConfigMapVolumeSource{
-					LocalObjectReference: corev1.LocalObjectReference{
-						Name: r.kbsConfig.Spec.KbsConfigMapName,
-					},
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: r.kbsConfig.Spec.KbsAuthSecretName,
 				},
 			},
 		})
 	}
-	if r.kbsConfig.Spec.KbsRvpsConfigMapName != "" {
-		// Check if the config map exists
-		foundConfigMap := &corev1.ConfigMap{}
+	return volumes, nil
+}
+
+func (r *KbsConfigReconciler) httpsConfigPresent() (bool, error) {
+	if r.kbsConfig.Spec.KbsHttpsKeySecretName == "" && r.kbsConfig.Spec.KbsHttpsCertSecretName == "" {
+		return false, nil
+	} else if r.kbsConfig.Spec.KbsHttpsKeySecretName != "" && r.kbsConfig.Spec.KbsHttpsCertSecretName != "" {
+		return true, nil
+	} else {
+		return false, std_errors.New("invalid https parameters, missing key or certificate")
+	}
+}
+
+func (r *KbsConfigReconciler) processHttpsSecret(ctx context.Context, volumes []corev1.Volume) ([]corev1.Volume, error) {
+	httpsConfigPresent, err := r.httpsConfigPresent()
+	if err != nil {
+		r.log.Error(err, "Failed to get KBS HTTPS secrets")
+		return nil, err
+	}
+	if httpsConfigPresent {
+		// get the https key and append to volumes
+		foundSecret := &corev1.Secret{}
 		err := r.Client.Get(ctx, client.ObjectKey{
 			Namespace: r.namespace,
-			Name:      r.kbsConfig.Spec.KbsRvpsConfigMapName,
-		}, foundConfigMap)
+			Name:      r.kbsConfig.Spec.KbsHttpsKeySecretName,
+		}, foundSecret)
 		if err != nil && errors.IsNotFound(err) {
-			// ConfigMap does not exist
-			r.log.Error(err, "KbsRvpsConfigMapName does not exist", "ConfigMap.Namespace", r.namespace, "ConfigMap.Name", r.kbsConfig.Spec.KbsRvpsConfigMapName)
-			return nil
+			r.log.Error(err, "KbsHttpsKeySecretName does not exist", "Secret.Namespace", r.namespace, "Secret.Name", r.kbsConfig.Spec.KbsHttpsKeySecretName)
+			return nil, err
 		} else if err != nil {
-			// Unknown error
-			r.log.Error(err, "Failed to get KBS RVPS ConfigMap")
-			return nil
+			r.log.Error(err, "Failed to get KBS HTTPS key Secret")
+			return nil, err
 		}
-		// ConfigMap exists
-		// Create the volume
+
 		volumes = append(volumes, corev1.Volume{
-			Name: "rvps-config",
+			Name: "https-key",
 			VolumeSource: corev1.VolumeSource{
-				ConfigMap: &corev1.ConfigMapVolumeSource{
-					LocalObjectReference: corev1.LocalObjectReference{
-						Name: r.kbsConfig.Spec.KbsRvpsConfigMapName,
-					},
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: r.kbsConfig.Spec.KbsHttpsKeySecretName,
+				},
+			},
+		})
+		// get the https key and append to volumes
+		foundSecret = &corev1.Secret{}
```

Comment:
Duplicate ?

Reply:
The `foundSecret` obj is a struct pointer that can be updated with the response returned by the Server. What is going to happen when calling `r.Client.Get` for the second time if the structure has been already populated? I'm not sure

Reply:
I think it'll be great if you rename the vars to foundHttpsKeySecret and foundHttpsCertSecret. I missed that earlier ;)